### DISCUSSION TITLE
LTSP in school - help on network and server

### DISCUSSION THREAD

**Discussion by jcarola** [ltsp/ltsp#821] [Q&A]
Hi to all!

We want to implement LTSP - Xubuntu in our school, around 100 PCs total across various classrooms.
We will start in our IT lab, library and classrooms teachers PCs (around 30 PCs) and expand as we can afford.

Tests:
I've been doing some tests in our IT lab (16 PCs) using a VM on my laptop (1 core from 11800H, 4GB RAM, 40 GB out of NVME Gen4, connected via 1Gb to the switch) as the server.
The test switch is an old 24 port 100Mb + 2 port 1Gb.
With the server and 1 client connected to the 2 ports 1Gb the performance is great!
Connecting all 16 clients to 100Mb and I begin to see some issues: if I use only one client the performance remains pretty good but, when using multiple clients simply opening Google Chrome takes a hit. Still pretty acceptable but, non the less, a pretty good hit! Storage and network seems to be the bottleneck.
And I'm testing 16 PCs but thinking about 100...

Network:
I was thinking about buying one Mikrotik CRS354-48G-4S+2Q+RM so we could connect each client via 1Gb ethernet, the server via multiple 10Gb, and connect other Mikrotiks via 40Gb when we can afford to implement in the classrooms.
Since we want this to be relatively future proof, would a 10Gb switch (they are so much more expensive!!!) make more sense so we could connect all clients via 10Gb, or would it be overkill?

Server:
I'm completely lost here... From my testing it seems:
Almost no RAM is needed.
I've seen CPU spikes when 16 machines are working, but I'm only allocating 1 core to the VM.
Would one, or multiple, NVME Gen 4 RAID 0 (ex: HighPoint SSD7505) make sense? Or distribute home folders across multiple "individual" NVME Gen 4?
Taking into account that multiple NVME Gen 4 (or RAID/RAIDs) and multiple 10Gb ports card will be needed, a CPU/Motherboard with some serious PCIe 4.0 lanes count would be needed... AMD Epyc 7252 or 7282 seems to be the cheapest option with 128 PCie 4.0 lanes...

I would very much appreciate your advice on all of this...
Please excuse my poor English.

Thank you.
João

**Comment by alkisg** [comment]
Hi João, I'll just mention some quick suggestions,

- I think 1 Gbps for the clients will be enough for a while. So your selected switch sound great.
- It's indeed possible that switches will need to be updated in the future, but they're too expensive to buy faster switches now. I think it's best to wait until 2.5 Gbps is widely available; e.g. maybe when you next update your workstations, they'll come with onboard 2.5 Gbps NICs. While cheap 10G NICs and switches will be more far away in the future.
- Don't put the LTSP server in a VM when you have so many users. Use bare metal. The bottleneck is /home, not the rootfs.
- ZFS for /home is a good idea, including snapshots (10minutely/hourly/daily/weekly/monthly). I'd use as many NVME/SSD disks as possible in RAID-0 mode for speed, and periodically (e.g. after work hours) rsync everything to another ZFS, snapshotted, 10 TB HDD connected in a simple SATA slot.

**Comment by jcarola** [comment]
Thank you for taking your time to help me Alkisg. I know these are just words but I really appreciate.

- So, you would keep rootfs in a single NVME and home in a NVME RAID 0?
- From what I've been reading, NVME RAID 0 has no big improvement in small random read/write, and have higher latency. That's why why I was stuck between NVME RAID 0 vs multiple individual NVME drives.
- Do you have any advice on server hardware?
I'm looking into AMD EPYC 7282 + ASRock ROMED8-2T/BCM + ASUS Hyper M.2 X16 PCIe 4.0 + 4 x Samsung 990 PRO 1TB. The biggest problem with this setup (that I have noticed) is that you can't boot from the RAID... This seems to be the cheapest option with room to grow (though it is a LOT of money... around 2.100€, plus case, RAM, PSU, etc.)
If RAID boot is a must, AMD 3955wx + Asus Pro WS WRX80E-SAGE SE seems to be the next choice at the cost of +900€...

Thanks again for your pacience!

**Comment by alkisg** [comment]
Hi João, for the rootfs, if you want fault tolerance, I suggest that you buy two small SATA SSDs (e.g. 250GB each) and install proxmox in RAID1 mode. Proxmox can create and auto-update a cloned UEFI partition, plus a ZFS rootfs RAID1 partition. This means that the motherboard doesn't need to support RAID at all, it's all software RAID, not hardware RAID, and the EFI partition is just copied as files, it's not using RAID at all.
Proxmox will also give you the ability to create a template VM and manage the LTSP image graphically over the browser, from any workstation inside the school.
So the result is that the server hardware doesn't need to be anything special.

This post that you have started, gave me the following idea about sponsorware guides, have a look: https://github.com/ltsp/ltsp/discussions/823

**Comment by jcarola** [comment]
Again, thank you for taking your time!

It's a great idea but I believe you forgot "_All of the above_" in the _about sponsorware guides_ post!!!
Also, if possible, there should be guides dedicated to server and network hardware, divided into use cases, with tried and tested setups. Setting up LTSP is relatively easy for a small project, you can even grab old hardware you have laying around and it will be just fine.
When you start thinking on a large scale, and you don't already own the hardware, it is a pain to blindly buy it, and this kind of help will not only save you time, also money and regrets!
The theoretical math seems simple: 40 clients at 1Gb = +-40Gb = 5000MB/s
One can say "no one will use max specs at the same time, let alone all time". Well... "prepare for the worst, hope for the best"
So, for top load, one would need, at least: One 48x 1Gb port + 4x 10Gb port switch, 1x NVME Gen 4.0 and 4x 10Gb port card.
But this is just math... What switch? Unmanaged vs L2 vs L3? Connect switch to switch or server to multiple switches? LACP? How to restrict connections to a list of MACs? What SSD? RAID or not to RAID? SSD performance with multiple simultaneous requests? What ethernet card? Preparing for scalability regarding PCIe lanes? What about CPU and RAM? etc., etc., etc....

**Comment by alkisg** [comment]
> It's a great idea but I believe you forgot "All of the above" in the about sponsorware guides post!!!

You're right, so I deleted the poll and I created the #823 discussion instead.
